File: testcases/x86_testcases/x86_output/test10/test10.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Define Employee structure
typedef struct {
    const char* name;
    const char* emp_id;
    const char* department;
} Employee;

// Define Manager structure
typedef struct {
    Employee super;
    Employee technician1;
    Employee technician2;
    int under;
} Manager;

// Define Technician structure
typedef struct {
    Employee super;
    const Employee* manager;
} Technician;

// Function to display employee information
void display_info(Employee* emp) {
    printf("Name: %s\n", emp->name);
    printf("Employee ID: %s\n", emp->emp_id);
    printf("Department: %s\n", emp->department);
}

// Function to display team information
void display_team_info(Manager* manager) {
    display_info(&manager->super);
    printf("Technicians under this manager:\n");
    display_info(&manager->technician1);
    printf("\n");
    display_info(&manager->technician2);
    printf("\n");
}

int main() {
    // Creating employees
    Manager manager1;
    manager1.super.name = "John Doe";
    manager1.super.emp_id = "M001";
    manager1.super.department = "Management";
    manager1.technician1.name = "";
    manager1.technician1.emp_id = "";
    manager1.technician1.department = "";
    manager1.technician2.name = "";
    manager1.technician2.emp_id = "";
    manager1.technician2.department = "";
    manager1.under = 0;

    Technician technician1;
    technician1.super.name = "Alice Smith";
    technician1.super.emp_id = "T001";
    technician1.super.department = "IT";
    technician1.manager = &manager1.super;

    Technician technician2;
    technician2.super.name = "Bob Johnson";
    technician2.super.emp_id = "T002";
    technician2.super.department = "IT";
    technician2.manager = &manager1.super;

    Manager manager2;
    manager2.super.name = "Jane Smith";
    manager2.super.emp_id = "M002";
    manager2.super.department = "HR";
    manager2.technician1.name = "";
    manager2.technician1.emp_id = "";
    manager2.technician1.department = "";
    manager2.technician2.name = "";
    manager2.technician2.emp_id = "";
    manager2.technician2.department = "";
    manager2.under = 0;

    Technician technician3;
    technician3.super.name = "Carol Williams";
    technician3.super.emp_id = "T003";
    technician3.super.department = "HR";
    technician3.manager = &manager2.super;

    // Adding technicians to managers
    manager1.technician1 = technician1.super;
    manager1.under = manager1.under + 1;
    manager1.technician2 = technician2.super;
    manager1.under = manager1.under + 1;

    manager2.technician1 = technician3.super;
    manager2.under = manager2.under + 1;

    // Displaying information
    display_team_info(&manager1);
    display_team_info(&manager2);

    return 0;
}

File: testcases/x86_testcases/test10.py
class Employee:
    def __init__(self, name : str, emp_id : str, department : str) -> None:
        self.name : str = name
        self.emp_id : str = emp_id
        self.department : str = department
        return

    def display_info(self) -> None:
        print("Name: ", self.name)
        print("Employee ID: ", self.emp_id)
        print("Department: ", self.department)
        return

class Manager(Employee):
    def __init__(self, name : str, emp_id : str, department : str) -> None:
        Employee.__init__(self, name, emp_id, department)
        self.technician1 : Employee = Employee("", "", "")
        self.technician2 : Employee = Employee("", "", "")
        self.under : int = 0
        return

    def add_technician(self, technician : Employee) -> None:
        if self.under == 0:
            self.technician1 = technician
            self.under = self.under + 1
        elif self.under == 1:
            self.technician2 = technician
            self.under = self.under + 1
        else:
            print("No more space for any technician under manager ", self.name)
        return

    def display_team_info(self) -> None:
        Employee.display_info(self)
        print("Technicians under this manager: ")
        technician1 : Employee = (self.technician1)
        technician1.display_info()
        print()
        technician2 : Employee = (self.technician2)
        technician2.display_info()
        print()
        return

class Technician(Employee):
    def __init__(self, name : str, emp_id : str, department : str, manager : Employee) -> None:
        Employee.__init__(self, name, emp_id, department)
        self.manager : Employee = manager
        return

    def display_info(self) -> None:
        Employee.display_info(self)
        manager : Employee = self.manager
        manager_name : str = manager.name
        print("Manager: ", manager_name)
        return

# Example usage:
if __name__ == "__main__":
    # Creating employees
    manager1 = Manager("John Doe", "M001", "Management")
    technician1 = Technician("Alice Smith", "T001", "IT", manager1)
    technician2 = Technician("Bob Johnson", "T002", "IT", manager1)

    manager2 = Manager("Jane Smith", "M002", "HR")
    technician3 = Technician("Carol Williams", "T003", "HR", manager2)

    # Adding technicians to managers
    manager1.add_technician(technician1)
    manager1.add_technician(technician2)
    manager2.add_technician(technician3)

    # Displaying information
    manager1.display_team_info()
    manager2.display_team_info()
